Storage/src/Connection/Rest.php

```diff
@@ -439,4 +443,30 @@ private function buildDownloadObjectParams(array $args)
             $requestOptions
         ];
     }
+
+    /**
+     * Generate a CRC32c checksum from a stream.
+     *
+     * @param StreamInterface $data
+     * @return string
+     */
+    private function crcFromStream(StreamInterface $data)
+    {
+        $pos = $data->tell();
+
+        if ($pos > 0) {
+            $data->rewind();
+        }
+
+        $crc32c = CRC32::create(CRC32::CASTAGNOLI);
+
+        $data->rewind();
+        while (!$data->eof()) {
+            $crc32c->update($data->read(1048576));
+        }
+
+        $data->seek($pos);
```

Comment:
Why does it seek back to the original `$pos` after rewinding to generate the hash?

Reply:
This function is based on the [guzzle `hash()`](https://github.com/guzzle/psr7/blob/master/src/functions.php#L410-L430) function. @dwsupplee do you think we need to reset the position in this situation?

Reply:
Interesting, I'm wondering why does it rewind a stream and not expect a user to provide a stream at the correct position? 

I think it's a nicety to verify the cursor position, but it's not very clear that this happens from the surface documentation.

Reply:
I like that this mirrors what we've already been doing, and think we should keep this as is.